### DISCUSSION TITLE
Application status is always Progressing when programmatically instantiated Pods are present

### DISCUSSION THREAD

**Discussion by cmpilato** [argoproj/argo-cd#15241] [Q&A]
_I'm a confessing Kubernetes/ArgoCD newbie, asking what I hope is not something I managed to overlook on page 1 of the docs, but..._

My application is a mixed bag in terms of k8s resource declaration. Most of the application is explicitly defined by k8s YAML files (Ingresses, Deployments, Services, etc.).  But a portion of the application is dynamic, Pods created _by code running in the application itself_ from PodTemplates defined in those same k8s YAML files.  It is this aspect of the application that I believe is causing my application to constantly appear in a "Progressing" state in ArgoCD.

Are there any best practices to be aware of as regards ArgoCD application health determination in such a scenario?

**Comment by crenshaw-dev** [comment]
> I'm a confessing Kubernetes/ArgoCD newbie

Welcome aboard, we're all newbies here in one way or another! 😄 

Can you post a screenshot of your app's resource tree in the Argo CD UI? I'd be surprised if a programmatically-generated Pod were keeping the app stuck in Progressing. Some top-level resource (i.e. some resource defined in git) must be Progressing, so we just gotta find that one and determine why it's not going Healthy.

**Reply by cmpilato** [reply]
A screenshot of the whole tree is not going to be possible (employer's IP, etc.).  But perhaps another piece of information I forgot to mentioned will help:

These dynamically created Pods are not permanent, but run for a while and then self-terminate.  I now realize as I look more closely at my ArgoCD UI tree that it's not the currently running Pods that are Progressing, but rather those which have been instantiated, run their course, and then self-terminated.  Here's a screenshot from the Details view of one such Pod:

![image](https://github.com/argoproj/argo-cd/assets/1846920/4c2486ff-8aab-4294-91a2-079bdda3d187)

**Reply by crenshaw-dev** [reply]
I think the tool which is spinning up these Pods must be adding the Argo CD tracking label and/or annotation. That's causing Argo CD to think, "ah, this used to be a resource represented in git, but now it's been removed from git." That's why the Pod is marked as "requires pruning."

You have a couple options. Either:
1) change the programmatic Pod-spinny-uppy tool to _not_ attach the Argo CD tracking label and/or annotation to the Pod
2) explicitly [mark the Pod as extraneous by adding an annotation](https://argo-cd.readthedocs.io/en/latest/user-guide/compare-options/#ignoring-resources-that-are-extraneous)

**Reply by cmpilato** [reply]
So long as I can find a way to work the term "Pod-spinny-uppy tool" into our documentation, I'll give these options a try.  Thanks!

**Reply by cmpilato** [reply]
Hrm... The following is already present on these Pods:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    argocd.argoproj.io/compare-options: IgnoreExtraneous
    argocd.argoproj.io/sync-options: Prune=false
...
```

**Reply by brhodes-pcty** [reply]
Hi @cmpilato -- did you ever find a solution for this? We're dealing with a similar challenge at the moment. Everything works, but the listener pod just sits at progressing, forever.

**Reply by cmpilato** [reply]
Unfortunately, I did not.
